lms/views/decorators/h_api.py

```diff
@@ -76,7 +76,7 @@ def wrapper(context, request):
         hapi_svc = request.find_service(name="hapi")
 
         is_instructor = any(
-            role in request.params["roles"].lower()
+            role in request.lti_user.roles.lower()
```

Comment:
So this line will now work for authorization param-authenticated requests (where there's no `request.params["roles"]`) as well as for actual launch requests